tests/sensors/test_external_task_sensor.py

```diff
@@ -576,6 +576,19 @@ def test_external_task_sensor_waits_for_dag_check_existence(self):
             op.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE, ignore_ti_state=True)
 
 
+def test_external_task_sensor_check_zipped_dag_existence(dag_maker):
+    with dag_maker(dag_id="external_task_sensor_dag"):
+        op = ExternalTaskSensor(
+            task_id="external_task_sensor",
+            external_dag_id="test_zip_dag",
+            external_task_id="dummy",
+            check_existence=True
+        )
+
+    with create_session() as session:
+        op._check_for_existence(session)
```

Comment:
We will need a real zipped dag to test this. You can create one at `tests/dags`

Reply:
It’s better to use a fixture to dynamically create/delete a zip file when tests are run and instead; a static zip file in `tests/dags` is opaque and easy to miss when we do refactoring. (IIRC this happened a while ago.)

See tests in `tests/models/test_dagbag.py` for some examples for this.